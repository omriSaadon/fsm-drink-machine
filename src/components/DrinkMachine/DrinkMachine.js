import React from 'react';
import DrinkChooser from 'components/DrinkChooser/DrinkChooser';
import Budget from 'components/Budget/Budget';
import DrinkFlowChart from 'components/DrinkFlowChart/DrinkFlowChart';
import OrderStatus from 'components/OrderStatus/OrderStatus';
import Payment from 'components/Payment/Payment';
import FsmStoreProvider from 'store/FsmStore';
import machineSpec from 'constants/machine-specification';
import NotificationStatus from 'components/Notifications/NotificationStatus';
import CardWrapper from 'components/CardWrapper/CardWrapper';
import classes from './DrinkMachine.scss';

const DrinkMachine = () => {
  const components = [Payment, Budget, DrinkChooser, OrderStatus];
  return (
    <FsmStoreProvider machineSpec={machineSpec}>
      <NotificationStatus />
      <div className={classes.wrapper}>
        <img className={classes.img} alt="drink" src={`${process.env.PUBLIC_URL}/assets/images/drink.jpg`} />
        <div className={classes.cards}>
          {components.map((Component) => <CardWrapper key={Component.name}><Component /></CardWrapper>)}
        </div>
      </div>
      <DrinkFlowChart />
    </FsmStoreProvider>
  );
};

export default DrinkMachine;
