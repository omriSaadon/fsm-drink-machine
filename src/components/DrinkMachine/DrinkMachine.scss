$small: 1150px;

.wrapper {
  display: flex;
}

.img {
  height: 650px;
  width: 450px;
  margin-right: 50px;
}

.cards {
  display: flex;
  flex-wrap: wrap;
  height: 550px;
}

@media (max-width: $small) {
  .cards {
    height: 1100px;
  }
}
