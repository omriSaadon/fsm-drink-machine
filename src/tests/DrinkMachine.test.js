import React from 'react';
import DrinkMachine from 'components/DrinkMachine/DrinkMachine';
import { machineRender } from './helpers/machineHelper';

jest.mock('react-flow-renderer', () => ({
  default: () => <div>Mock test flow</div>,
  __esModule: true,
}));

test('Check Insufficient funds', () => {
  const { asFragment } = machineRender(
    <DrinkMachine />,
  );
  expect(asFragment()).toMatchSnapshot();
});
