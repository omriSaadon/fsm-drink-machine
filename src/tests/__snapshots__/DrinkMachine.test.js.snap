// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Insufficient funds 1`] = `
<DocumentFragment>
  <div />
  <div>
    <img
      alt="drink"
      src="/assets/images/drink.jpg"
    />
    <div>
      <div>
        <div
          class="ant-card ant-card-bordered"
        >
          <div
            class="ant-card-head"
          >
            <div
              class="ant-card-head-wrapper"
            >
              <div
                class="ant-card-head-title"
              >
                Payment
              </div>
            </div>
          </div>
          <div
            class="ant-card-body"
          >
            <div
              class="ant-alert ant-alert-warning ant-alert-no-icon"
              data-show="true"
              role="alert"
            >
              <div
                class="ant-alert-content"
              >
                <div
                  class="ant-alert-message"
                >
                  We accepts coins only
                </div>
                <div
                  class="ant-alert-description"
                />
              </div>
            </div>
            <div>
              <button
                class="ant-btn ant-btn-round"
                type="button"
              >
                <span>
                  1 ₪
                </span>
              </button>
              <button
                class="ant-btn ant-btn-round"
                type="button"
              >
                <span>
                  2 ₪
                </span>
              </button>
              <button
                class="ant-btn ant-btn-round"
                type="button"
              >
                <span>
                  5 ₪
                </span>
              </button>
              <button
                class="ant-btn ant-btn-round"
                type="button"
              >
                <span>
                  10 ₪
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="ant-card ant-card-bordered"
        >
          <div
            class="ant-card-head"
          >
            <div
              class="ant-card-head-wrapper"
            >
              <div
                class="ant-card-head-title"
              >
                Wallet
              </div>
            </div>
          </div>
          <div
            class="ant-card-body"
          >
            <div>
              Current Budget: 10 ₪
            </div>
            <div>
              Next Deposit: 10
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="ant-card ant-card-bordered"
        >
          <div
            class="ant-card-head"
          >
            <div
              class="ant-card-head-wrapper"
            >
              <div
                class="ant-card-head-title"
              >
                Choose Your Drink
              </div>
            </div>
          </div>
          <div
            class="ant-card-body"
          >
            <div>
              <div>
                <button
                  class="ant-btn ant-btn-round ant-btn-lg"
                  type="button"
                >
                  <img
                    alt=""
                    src="/assets/images/coca-cola.jpeg"
                  />
                  <span>
                    Coke - 8 ₪
                  </span>
                </button>
                <button
                  class="ant-btn ant-btn-round ant-btn-lg"
                  type="button"
                >
                  <img
                    alt=""
                    src="/assets/images/sparkling-water.jpeg"
                  />
                  <span>
                    Sparkling Water - 6 ₪
                  </span>
                </button>
                <button
                  class="ant-btn ant-btn-round ant-btn-lg"
                  type="button"
                >
                  <img
                    alt=""
                    src="/assets/images/water.jpeg"
                  />
                  <span>
                    Water - 5 ₪
                  </span>
                </button>
                <button
                  class="ant-btn ant-btn-round ant-btn-lg"
                  type="button"
                >
                  <img
                    alt=""
                    src="/assets/images/ice-tea.jpeg"
                  />
                  <span>
                    Ice Tea - 7 ₪
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="ant-card ant-card-bordered"
        >
          <div
            class="ant-card-head"
          >
            <div
              class="ant-card-head-wrapper"
            >
              <div
                class="ant-card-head-title"
              >
                Order Status
              </div>
            </div>
          </div>
          <div
            class="ant-card-body"
          >
            You have paid 0 ₪
            <button
              class="ant-btn ant-btn-round ant-btn-sm"
              type="button"
            >
              <span>
                Cancel Payment
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div>
      Mock test flow
    </div>
  </div>
</DocumentFragment>
`;
